datajunction-server/datajunction_server/internal/nodes.py

```diff
@@ -830,138 +836,60 @@ async def propagate_update_downstream(  # pylint: disable=too-many-locals
-                        joinedload(Node.current).options(
-                            selectinload(NodeRevision.columns).options(
-                                selectinload(Column.node_revisions).options(
-                                    selectinload(NodeRevision.node),
-                                ),
-                            ),
-                            selectinload(NodeRevision.cube_elements).options(
-                                selectinload(Column.node_revisions).options(
-                                    selectinload(NodeRevision.node),
-                                ),
-                            ),
-                        ),
-                    ],
-                )
-                if child_node:
-                    await update_cube_node(
-                        session,
-                        child_node.current,  # type: ignore
-                        UpdateNode(
-                            metrics=[
-                                metric.name for metric in child_node.current.cube_metrics()  # type: ignore  # pylint: disable=line-too-long
-                            ],
-                            dimensions=child_node.current.cube_dimensions(),  # type: ignore
-                        ),
-                        query_service_client=query_service_client,
-                        background_tasks=background_tasks,
-                        validate_access=validate_access,
-                    )
-                continue
-
-            node_validator = await validate_node_data(
-                data=child,
-                session=session,
-            )
-            result = await Node.get_by_name(
-                session,
-                child.name,
-                options=[
-                    joinedload(Node.current).options(
-                        *NodeRevision.default_load_options()
-                    ),
-                ],
-            )
-            if not result:
-                continue
-            child = result.current  # type: ignore
-            # Update the child with a new node revision if its columns or status have changed
-            if node_validator.differs_from(child):
-                await session.refresh(child, ["required_dimensions"])
-                new_revision = copy_existing_node_revision(child)
-                new_revision.version = str(
-                    Version.parse(child.version).next_major_version(),
-                )
-
-                new_revision.status = node_validator.status
-                new_revision.lineage = [
-                    lineage.dict()
-                    for lineage in await get_column_level_lineage(session, new_revision)
-                ]
+    _logger.info("Propagating update of node %s downstream", node.name)
+    downstreams = await get_downstream_nodes(
+        session,
+        node.name,
+        include_deactivated=False,
+        include_cubes=False,
+    )
+    downstreams = topological_sort(downstreams)
```

Comment:
Thank you @shangyian for catching and fixing this. I completely missed this aspect of revalidating child nodes.